### PR TITLE
Search: replace ChartJS with Jetpack RecordMeterBar component in Record Meter

### PR DESCRIPTION
**Pull request by annacmc**
This is testing out switching the record meter bar components. This draft handles all the necessary data structure changes to switch the component.

Note: the design still would need some CSS work to look a bit better (especially the legend items) but it gives an overview of how the new component fits in and works with our data. 

image

**Pros:**

* Very small values with very large totals still look small. However, they don’t break the display. 

* The record count does not get provided an expected total. Instead, it is calculated from the combination of values (including the ‘remaining’ value) this means if you are over your limit, or have no limit, the chart just looks full. It is not affected by negative remaining values. 

* Legend is much more flexible & yes flex boxes nicely

**Cons:**

* We still can’t easily scale the data, as the numbers are now auto added to the legend. 
* The legend currently outputs the value before the label. 
* No longer ordered by count, we will need to add sorting into the data generation before passing to chart.

### CODE REVIEW

File: projects/packages/search/src/dashboard/components/record-meter/lib/record-info.js [hunk 1]
```diff
@@ -65,43 +65,43 @@ export default function getRecordInfo(
 		}
 
 		// sort & split items into included and other
-		const postTypeItems = splitUsablePostTypes(
+		const PostTypeItems = splitUsablePostTypes(
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/lib/record-info.js:68]: I think `postTypeItems` is correct here (unless PostTypeItems is a class)?

**Comment by bluefuton** [comment]
I'm pretty encouraged by the results here! We lose the hover behaviour but I don't miss it.

> Very small values with very large totals still look small. However, they don’t break the display.

The plus side is we could specify a `min-width: 10px` on each item in the chart and that might well solve it. It's not _quite_ a true representation of the data but it would mean every segment was visible.

Screen Shot 2022-05-13 at 17 42 05

> The legend currently outputs the value before the label.

We could fix this in the underlying component (or add a prop to vary the display if existing users of the component want to keep it as is).

> No longer ordered by count, we will need to add sorting into the data generation before passing to chart.

We could also change the underlying component to do this for us (again, optionally via a prop if existing users aren't keen on that being that default behaviour).

**Comment by bluefuton** [comment]
> I'm going to check in with the existing users of the component and figure out next steps from there for these items.

Excellent!

If we use it, one more thing we will want to add to the component is accessible output (like we've done in our own component outputting a data table inside the `<canvas>`).

File: projects/packages/search/src/dashboard/components/record-meter/lib/record-info.js (hunk 1 above)
Comment (kangzj) [reply projects/packages/search/src/dashboard/components/record-meter/lib/record-info.js:68]: I think so. We should use camelCase in JS.

image

REF: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/#naming-conventions

**Comment by a8ck3n** [comment]
Would it make sense to have an option for a non-flex legend?

I personally would prefer that as I'm not a fan of having to "scrub" a UI to read labels. This is something that my partner gets annoyed with as well as her `/wp-admin` is in Japanese and rarely do these truncated labels show enough to be sensible for her. I imagine other "longer" languages (like German?) would see the same issue with truncation.

**Comment by bluefuton** [comment]
Thanks for the feedback @a8ck3n! Sounds like it's worth us doing some testing in those languages to see what the experience is like.

**Comment by bluefuton** [comment]
Todo:
- [ ] add sort prop (relies on https://github.com/Automattic/jetpack/pull/24518)
- [ ] sort out a11y content in component

**Comment by bluefuton** [comment]
We'll remove the chart.js dependency in a future PR.

**Comment by bluefuton** [comment]
Ah - here's an interesting situation. The new sorting behaviour in RecordMeterBar means that 'Remaining' appears first:

Screen Shot 2022-06-03 at 11 58 25

**Review by bluefuton — DISMISSED** [review]

**Review by jsnmoon — APPROVED** [review]

**Pull request merged**